### PR TITLE
Add max_connections to OfferRequest creation

### PR DESCRIPTION
**Pull request by JamesWair**
This PR adds the [max_connections](https://duffel.com/docs/api/v1/offer-requests/create-offer-request#offer-requests-create-an-offer-request-body-parameters-max-connections) parameter to the OfferRequest create request.

### CODE REVIEW

File: duffel_api/api/booking/offer_requests.py [hunk 1]
```diff
@@ -85,6 +89,12 @@ def _validate_slices(slices):
             ):
                 raise OfferRequestCreate.InvalidSlice(travel_slice)
 
+    @staticmethod
+    def _validate_max_connections(max_connections):
+        """Validate the max connection number"""
+        if max_connections < 0:
```
Comment (ulissesalmeida) [inline duffel_api/api/booking/offer_requests.py:95]: we only support integers in the API, I wonder if we should:

```suggestion
        if isinstance(max_connections, int) and max_connections < 0:
```

Comment (JamesWair) [reply duffel_api/api/booking/offer_requests.py:95]: Good shout! Assuming `isinstance` returns True in this case when it is an int, I think we would want:
`if not isinstance(max_connections, int) or max_connections < 0:`
as we raise and error when True. I'll update now

Comment (sgerrand) [reply duffel_api/api/booking/offer_requests.py]: 👋 Should these kinds of validations occur in the client, e.g. should these validations occur here _and_ in the API?

Comment (cianyleow) [reply duffel_api/api/booking/offer_requests.py]: I'd agree with @sgerrand that we've generally avoided any sort of explicit type checking in the client and only performed basic consistency checks (i.e. that there are non-zero elements or certain attributes exist).

**Review by cianyleow — APPROVED** [review]
LGTM, I can see the value in having the `int` check to guide developers to using the correct numeric type.

We do perform this validation on the server side as well (passing in a `float` results in an error), so this is valid, but we should aim to roll this out consistently in the future.

**Pull request merged**